ibm/service/power/resource_ibm_pi_instance.go

```diff
@@ -678,18 +678,28 @@ func resourceIBMPIInstanceUpdate(ctx context.Context, d *schema.ResourceData, me
 	cores_enabled := checkCloudInstanceCapability(cloudInstance, CUSTOM_VIRTUAL_CORES)
 
 	if d.HasChanges(Arg_InstanceName, Arg_VirtualOpticalDevice) {
+		if d.HasChange(Arg_InstanceName) && d.HasChange(Arg_VirtualOpticalDevice) {
+			oldVOD, _ := d.GetChange(Arg_VirtualOpticalDevice)
+			d.Set(Arg_VirtualOpticalDevice, oldVOD)
+			return diag.Errorf("updates to %s and %s are mutually exclusive", Arg_InstanceName, Arg_VirtualOpticalDevice)
+		}
```

Comment:
Updates are mutually exclusive for name/active vm resources and cloud initialization, so an error will be thrown here